core/o11y/src/macros.rs

```diff
@@ -20,6 +20,28 @@ macro_rules! handler_span {
     }};
 }
 
+/// Creates a TRACE level span.
+#[macro_export]
+macro_rules! handler_trace_span {
+    ($target:expr, $msg:expr $(, $extra_fields:tt)*) => {{
+        let WithSpanContext { msg, context, .. } = $msg;
+
+        let actor = near_o11y::macros::last_component_of_name(std::any::type_name::<Self>());
+        let handler = near_o11y::macros::type_name_of(&msg);
+
+        let span = tracing::trace_span!(
+            target: $target,
+            "handle",
+            handler,
+            actor,
+            $($extra_fields)*)
+        .entered();
+
+        span.set_parent(context);
+        (span, msg)
+    }};
+}
```

Comment:
Instead of duplicating the macro body, consider making `handler_span` able to take a parameter that describes level (and invokes `tracing::span!`) and _then_ add `handler_debug_span`/`handler_trace_span` helpers.

The matcher will likely look a lot like this:

```rust
macro_rules! handler_span {
    (target: $target:expr, $lvl:expr, $msg:expr, $($fields:tt)*) => {{
        let WithSpanContext { msg, context, .. } = $msg;
        let span = tracing::span!(
            target: $target,
            $lvl,
            "handle",
            handler = near_o11y::macros::type_name_of(&msg),
            actor = near_o11y::macros::last_component_of_name(std::any::type_name::<Self>()),
            $($extra_fields)*)
        .entered();
        span.set_parent(context);
        (span, msg)
    }}
}

macro_rules! handler_trace_span {
    ($target:expr, $msg:expr, $($extra_fields:tt)*) => {
        $crate::handler_span!(target: $target, $crate::tracing::Level::TRACE, $msg, $($extra_fields)*
    }
}
```

I see a couple issues I didn’t spot last time too:

1. it is worth it to keep `actor` and `handler` expressions within the `span!` macro, as this way they will end up within the `if ... {}` condition that decides whether the span is enabled at all, and will be guaranteed to not be processed if the span is disabled;
2. The `extra_fields` `tt` matcher should not be separated by commas, I think. It prevents passing through valid token forests such as `foo = expression()` etc.

Reply:
Followed the approach from https://docs.rs/tracing/latest/src/tracing/macros.rs.html#24  .
Tested that both the `handler_trace_span!` can be called with repeated extra fields, including `a = 2+3` fields.